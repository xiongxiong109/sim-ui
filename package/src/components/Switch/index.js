import Switch from './src/main'

export default Switch