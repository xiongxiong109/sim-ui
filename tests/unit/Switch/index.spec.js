// Switch 组件测试
import { mount } from '@vue/test-utils'
import Page from './test'
import Switch from '@/components/Switch'

const factory = () => {
  const wrapper = mount(Page);
  const smSwitch = wrapper.find(Switch);
  const switchBtn = smSwitch.find('div');
  return {
    wrapper,
    smSwitch,
    switchBtn
  }
}

describe('Switch Component', () => {

  it('isDisabled控制开关行为', () => {

    const { wrapper, smSwitch } = factory();

    wrapper.setData({
      isOpen: false,
      isDisabled: false
    })

    expect(smSwitch.vm.isDisabled).toBe(false);
  })

  it('正常状态下点击, v-model控制双向数据变化', () => {

    const { wrapper, smSwitch, switchBtn } = factory();
    
    wrapper.setData({
      isOpen: false,
      isDisabled: false
    })

    switchBtn.trigger('click');
    // 触发input
    expect(smSwitch.emitted('input')).toBeTruthy();
    // v-model
    expect(smSwitch.vm.value).toBe(true)
    expect(wrapper.vm.isOpen).toEqual(smSwitch.vm.value);
    // 确保UI的快照, snapshot
    expect(wrapper.element).toMatchSnapshot()
  })

  it('禁用状态下点击无效', () => {

    const { wrapper, smSwitch, switchBtn } = factory();

    wrapper.setData({
      isDisabled: true,
      isOpen: false
    });

    switchBtn.trigger('click');
    // 不触发input事件
    expect(smSwitch.emitted('input')).toBeFalsy();
    // 数据无变化
    expect(wrapper.vm.isOpen).toEqual(smSwitch.vm.value);
    expect(wrapper.element).toMatchSnapshot()

  })

})