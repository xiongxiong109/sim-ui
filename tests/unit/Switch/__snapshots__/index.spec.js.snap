// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch Component 正常状态下点击, v-model控制双向数据变化 1`] = `
<div
  class="sm-switch__wrapper is-open"
/>
`;

exports[`Switch Component 禁用状态下点击无效 1`] = `
<div
  class="sm-switch__wrapper"
/>
`;
